content/en/docs/components/pipelines/author-a-pipeline/components.md

```diff
@@ -0,0 +1,222 @@
+
+## Summary
+A *component* is the basic unit of execution logic in KFP. A component is a named template for how to run a container using an image, a command, and arguments. Components may also have inputs and outputs, making a component a computational template, analogous to a function.
+
+Component inputs are dynamic data used in either the container commands or arguments.
+
+Component outputs may be machine learning artifacts or other JSON-serializable data.
+
+## Author a component
+
+At the lowest level of execution, all components define their execution logic via a container image, command, and arguments. An importer component is a special case and the only exception to this.
+
+The KFP SDK exposes three ways of authoring components with these three properties.
+
+### 1. Lighweight Python function-based components
+The most simple way to author a component is via a lightweight Python function-based component (also known as a lightweight component).
+
+Lightweight components provides a fully Pythonic approach to creating a component that executes a single Python function within a container at runtime.
+
+To create a lightweight component, you must:
+1. Define a standalone function.
+
+    A standalone Python function is a function that does not reference any symbols defined outside of its scope. This means the function must define all objects it uses within the scope of the function and must include all import statements within the function body.
+
+2. Include type annotations for the function parameters and return values.
+
+    Type annotations indicate what the component inputs and outputs are and tells the KFP lightweight component executor how to serialize and deserialize the data as it is passed within a pipeline. This also (optionally) allows the KFP DSL compiler to type check your pipeline.
+
+    Valid parameter annotations include `str`, `int`, `float`, `bool`, `dict`, `list`, `OutputPath`, `InputPath`, `Input[<Artifact>]`, and `Output[<Artifact>]`.
+    
+    Valid return annotations include `str`, `int`, `float`, `bool`, `dict`, and `list`. You may also specify multiple return values by using these annotations within a `typing.NamedTuple`.
+
+    For detailed discussion on type annotations and runtime behavior, see [Data Passing](/).
+
+3. Decorate your function with the `@kfp.dsl.component` decorator.
+    This decorator transforms a Python function into a component that can be used within a pipeline.
+
+    For a comprehensive list of `@kfp.dsl.component` decorator arguments, see the DSL [reference documentation](https://kubeflow-pipelines.readthedocs.io/en/master/source/dsl.html).
+
+
+The following is an example of a lightweight component that trains a model on an existing input `Dataset` artifact for `num_epochs` epochs, then saves the output `Model` artifact.
+
+```python
+from kfp import dsl
+
+@dsl.component(
+    base_image='python:3.7',
+    packages_to_install=['tensorflow']
```

Comment:
nit: may consider specifying a version of packages to install (`tensorflow==1.3.0`) and mentioning that these are pip installed.